### PR TITLE
docdb deletion protection

### PR DESCRIPTION
**Pull request by muzzamilinovaqo**

### CODE REVIEW

File: plugins/aws/documentDB/docdbDeletionProtectionEnabled.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'DocumentDB Cluster Deletion Protection',
+    category: 'DocumentDB',
+    domain: 'Databases',
+    severity: 'Medium',
+    description: 'Ensure that your Amazon DocumentDB clusters have deletion protection feature enabled.',
```
Comment (fatima99s) [inline plugins/aws/documentDB/docdbDeletionProtectionEnabled.js:9]:
```suggestion
    description: 'Ensure that Amazon DocumentDB clusters have deletion protection feature enabled.',
```

File: plugins/aws/documentDB/docdbDeletionProtectionEnabled.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    more_info: 'Enabling deletion protection feature for your Amazon DocumentDB clusters acts as a safety net, preventing accidental deletions and ensuring your data stays secure and accessible at all times.',
```
Comment (fatima99s) [inline plugins/aws/documentDB/docdbDeletionProtectionEnabled.js:10]:
```suggestion
    more_info: 'Enabling deletion protection feature for Amazon DocumentDB clusters acts as a safety net, preventing accidental deletions and ensuring that the data stays secure and accessible at all times.',
```

File: plugins/aws/documentDB/docdbDeletionProtectionEnabled.spec.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                expect(results[0].region).to.equal('us-east-1');
+                done();
+            });
+        });
+
+        it('should UNKNOWN if unable to list DocumentDB Clusters', function (done) {
+            const cache = createCache(null,  { message: "Unable to list DocumentDB Clusters encryption" });
+            docdbDeletionProtectionEnabled.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(3);
```
Comment (fatima99s) [inline plugins/aws/documentDB/docdbDeletionProtectionEnabled.spec.js:79]: add the message

**Review by fatima99s — CHANGES_REQUESTED** [review]

File: plugins/aws/documentDB/docdbDeletionProtectionEnabled.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'DocumentDB Cluster Deletion Protection',
+    category: 'DocumentDB',
+    domain: 'Databases',
+    severity: 'Medium',
+    description: 'Ensure that Amazon DocumentDB clusters have deletion protection feature enabled.',
+    more_info: 'Enabling deletion protection feature for Amazon DocumentDB clusters acts as a safety net, preventing accidental deletions and ensuring that the data stays secure and accessible at all times.',
```
Comment (alphadev4) [inline plugins/aws/documentDB/docdbDeletionProtectionEnabled.js:10]:
```suggestion
    more_info: 'Enabling deletion protection feature for Amazon DocumentDB clusters acts as a safety net, preventing accidental database deletions or deletion by an unauthorized user.  A Neptune DB cluster can't be deleted while deletion protection is enabled which ensures that the data stays secure and accessible at all times.',
```

File: plugins/aws/documentDB/docdbDeletionProtectionEnabled.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            if (!describeDBClusters.data.length) {
+                helpers.addResult(results, 0,
+                    'No DocumentDB clusters found', region);
+                return rcb();
+            }
+            
+            for (let cluster of describeDBClusters.data) {
+                if (!cluster.DBClusterArn) continue;
+
+                let resource = cluster.DBClusterArn;
```
Comment (alphadev4) [inline plugins/aws/documentDB/docdbDeletionProtectionEnabled.js:42]:
```suggestion
```

File: plugins/aws/documentDB/docdbDeletionProtectionEnabled.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+                if (cluster.DeletionProtection) {
+                    helpers.addResult(results, 0,
+                        'DocumentDB cluster has deletion protection enabled',
+                        region, resource);
```
Comment (alphadev4) [inline plugins/aws/documentDB/docdbDeletionProtectionEnabled.js:47]:
```suggestion
                        region, cluster.DBClusterArn);
```

Comment (alphadev4) [inline plugins/aws/documentDB/docdbDeletionProtectionEnabled.js:47]:
```suggestion
                    helpers.addResult(results, 0, 'DocumentDB cluster has deletion protection enabled', region, resource);
```

File: plugins/aws/documentDB/docdbDeletionProtectionEnabled.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+                } else {
+                    helpers.addResult(results, 2,
+                        'DocumentDB cluster does not have deletion protection enabled',
+                        region, resource);
```
Comment (alphadev4) [inline plugins/aws/documentDB/docdbDeletionProtectionEnabled.js:51]:
```suggestion
                    helpers.addResult(results, 2,  'DocumentDB cluster does not have deletion protection enabled', region, cluster.DBClusterArn);
```

File: plugins/aws/documentDB/docdbDeletionProtectionEnabled.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+            }
+            
+            for (let cluster of describeDBClusters.data) {
+                if (!cluster.DBClusterArn) continue;
+
+
+                if (cluster.DeletionProtection) {
+                    helpers.addResult(results, 0, 'DocumentDB cluster has deletion protection enabled', region, cluster.DBClusterArn);
+                } else {
+                    helpers.addResult(results, 2,  'DocumentDB cluster does not have deletion protection enabled', region, cluster.DBClusterArn);
```
Comment (alphadev4) [inline plugins/aws/documentDB/docdbDeletionProtectionEnabled.js:46]:
```suggestion
                    helpers.addResult(results, 2, 'DocumentDB cluster does not have deletion protection enabled', region, cluster.DBClusterArn);
```

File: plugins/aws/documentDB/docdbDeletionProtectionEnabled.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'DocumentDB Cluster Deletion Protection',
+    category: 'DocumentDB',
+    domain: 'Databases',
+    severity: 'Medium',
+    description: 'Ensure that Amazon DocumentDB clusters have deletion protection feature enabled.',
+    more_info: 'Enabling deletion protection feature for Amazon DocumentDB clusters acts as a safety net, preventing accidental database deletions or deletion by an unauthorized user.  A Neptune DB cluster can't be deleted while deletion protection is enabled which ensures that the data stays secure and accessible at all times.',
+    recommended_action: 'Modify DocumentDb cluster and enable deletion protection.',
```
Comment (alphadev4) [inline plugins/aws/documentDB/docdbDeletionProtectionEnabled.js:11]:
```suggestion
    more_info: 'Enabling deletion protection feature for Amazon DocumentDB clusters acts as a safety net, preventing accidental database deletions or deletion by an unauthorized user.  A Neptune DB cluster can\'t be deleted while deletion protection is enabled which ensures that the data stays secure and accessible at all times.',
    recommended_action: 'Modify DocumentDb cluster and enable deletion protection.',
```

File: plugins/aws/documentDB/docdbDeletionProtectionEnabled.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+            if (!describeDBClusters.data.length) {
+                helpers.addResult(results, 0,
+                    'No DocumentDB clusters found', region);
+                return rcb();
+            }
+            
+            for (let cluster of describeDBClusters.data) {
+                if (!cluster.DBClusterArn) continue;
+
+
```
Comment (alphadev4) [inline plugins/aws/documentDB/docdbDeletionProtectionEnabled.js:42]:
```suggestion
```

**Review by alphadev4 — APPROVED** [review]

File: plugins/aws/documentDB/docdbDeletionProtectionEnabled.js [hunk 11]
```diff
@@ -0,0 +1,54 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'DocumentDB Cluster Deletion Protection',
+    category: 'DocumentDB',
+    domain: 'Databases',
+    severity: 'Medium',
+    description: 'Ensure that Amazon DocumentDB clusters have deletion protection feature enabled.',
+    more_info: 'Enabling deletion protection feature for Amazon DocumentDB clusters acts as a safety net, preventing accidental database deletions or deletion by an unauthorized user.  A Neptune DB cluster can\'t be deleted while deletion protection is enabled which ensures that the data stays secure and accessible at all times.',
```
Comment (alphadev4) [inline plugins/aws/documentDB/docdbDeletionProtectionEnabled.js:10]:
```suggestion
    more_info: 'Enabling deletion protection feature for Amazon DocumentDB clusters acts as a safety net, preventing accidental database deletions or deletion by an unauthorized user. It ensures that the data stays secure and accessible at all times.',
```

**Pull request merged**